### PR TITLE
Update itc plot when conditions change

### PR DESCRIPTION
**Pull request by cquiroz**
This PR brings several improvements to the plots to reduce the amount of redraws and properly react to condition changes

### CODE REVIEW

File: model/shared/src/main/scala/explore/model/OdbItcResult.scala [hunk 1]
```diff
@@ -33,7 +34,10 @@ object OdbItcResult {
   ) extends OdbItcResult
       derives Eq {
     def toItcExposureTime: ItcExposureTime =
-      ItcExposureTime(OverridenExposureTime.FromItc, exposureTime, exposures)
+      ItcExposureTime(OverridenExposureTime.FromItc,
+                      exposureTime,
+                      PosInt.unsafeFrom(exposures.value)
```
Comment (toddburnside) [inline model/shared/src/main/scala/explore/model/OdbItcResult.scala:39]: How do you know this is OK?

**Review by toddburnside — APPROVED** [review]
LGTM

File: explore/src/main/scala/explore/itc/ItcProps.scala [hunk 2]
```diff
@@ -49,62 +50,56 @@ import queries.schemas.odb.ObsQueries.*
...
+  private val asterismIds   = obsSummary.scienceTargetIds
 
+  private val remoteConfig                   = obsSummary.observingMode.map { o =>
+    BasicConfigAndItc(o.toBasicConfiguration,
+                      remoteExposureTime.map(ItcResult.fromItcExposureTime(_).rightNec)
+    )
+  }
+  //
   // if there is an observingMode, that means a configuration has been created. If not, we'll use the
   // row selected in the spectroscopy modes table if it exists
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/itc/ItcProps.scala:73]: This is still valid I think, we use the remote mode or else the one in the table

File: model/shared/src/main/scala/explore/model/OdbItcResult.scala (hunk 1 above)
Comment (cquiroz) [reply model/shared/src/main/scala/explore/model/OdbItcResult.scala:39]: We logically know it is never 0
The next lucuma-core will reconcile the mismatching usages across PosInt and NonNegInt

File: explore/src/main/scala/explore/itc/ItcProps.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (toddburnside) [reply explore/src/main/scala/explore/itc/ItcProps.scala:73]: OK. I just noticed that there isn't an observing mode in the code.

[the remark this answers is not in the document]
Comment (cquiroz) [reply explore/src/main/scala/explore/itc/ItcProps.scala:73]: Right, I'll update the comment

**Pull request merged**